analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ObjectsShouldNotBeDisposedMoreThanOnce.vb

```diff
@@ -0,0 +1,242 @@
+﻿Imports System
+Imports System.IO
+Imports System.Data.Common
+Imports System.Threading.Tasks
+
+Interface IWithDispose
+    Inherits IDisposable
+End Interface
+
+Public Class Disposable
+    Implements IDisposable
+
+    Public Sub Dispose() Implements IDisposable.Dispose
+    End Sub
+
+End Class
+
+Public Class DisposableAlias
+    Implements IDisposable
+
+    Public Sub CleanUp() Implements IDisposable.Dispose
+    End Sub
+
+End Class
```

Comment:
I do not understand - the proposed change is an extra test to add or I need to change the existing one because it's not a valid case or so something else?

Reply:
Let's add a test case where as well like this
```
    Public Sub DisposedTwice_PrivateAlias_Using()
        Using d As New DisposablePrivateAlias()  ' Noncompliant
            DirectCast(d, IDisposable).Dispose()
        End Using
    End Sub
```